app/routes/application.js

```diff
@@ -7,19 +7,55 @@ export default Route.extend({
   beforeModel(transition) {
     const kredits = this.kredits;
 
-    return kredits.setup().then(() => {
-      kredits.get('kredits').preflightChecks().catch((error) => {
-        console.error('Kredits preflight check failed!');
-        console.error(error);
-      });
-      if (kredits.get('accountNeedsUnlock')) {
-        if (confirm('It looks like you have an Ethereum wallet available. Please unlock your account.')) {
-          transition.retry();
+    let selectedNetworkId = window.ethereum.networkVersion;
+    let kreditsNetwork = kredits.getKreditsDeployedNetwork();
+    let kreditsNetworkName;
+
+    switch(kreditsNetwork) {
```

Comment:
This belongs in a separate function or config. But from what I remember I'm pretty sure you can get network names from the `ether.js` library/API anyway, so there's no need to re-implement our own version of this.